src/tests/ftest/util/pool_security_test_base.py

```diff
@@ -0,0 +1,444 @@
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import os
+from apricot import TestWithServers
+from daos_utils import DaosCommand
+from dmg_utils import DmgCommand
+import dmg_utils
+import random
+import pwd
+import grp
+import re
+from general_utils import pcmd
+
+PERMISSIONS = ["", "r", "w", "rw"]
+
+def acl_entry(usergroup, name, permission):
+    '''
+    Deascription:
+        Create a daos acl entry on the specified user or groups
+        with specified or random permission.
+    Args:
+        usergroup: user or group.
+        name: user or group name to be created.
+        permission: permission to be created.
+    Return:
+        entry: daos pool acl entry.
+    '''
+    if permission == "random":
+        permission = random.choice(PERMISSIONS)
+    if permission == "nonexist":
+        return ""
+    if "group" in usergroup:
+        entry = "A:G:" + name + "@:" + permission
+    else:
+        entry = "A::" + name + "@:" + permission
+    return entry
+
+def add_del_user(hosts, ba_cmd, user):
+    '''
+    Deascription:
+        Add or delete the daos user and group on host by sudo command.
+    Args:
+        hosts: list of host.
+        ba_cmd: linux bash command to create user or group.
+        user: user or group name to be created or cleaned.
+    Return:
+        none.
+    '''
+    bash_cmd = os.path.join("/usr/sbin", ba_cmd)
+    homedir = ""
+    if "usermod" not in ba_cmd and "user" in ba_cmd:
+        homedir = "-r"
+    cmd = " ".join(("sudo", bash_cmd, homedir, user))
+    pcmd(hosts, cmd, False)
+
+def create_acl_file(file_name, permissions):
+    '''
+    Deascription:
+        Create a acl_file with permissions.
+    Args:
+        file_name: file name.
+        permissions: daos acl permission list.
+    Return:
+        none.
+    '''
+
+    acl_file = open(file_name, "w")
+    acl_file.write("\n".join(permissions))
+    acl_file.close()
+
+class PoolSecurityTestBase(TestWithServers):
+    '''
+    Test Class Description:
+        Test methods to verify the Pool security with acl by dmg
+        and daos tools.
+    :avocado: recursive
+    '''
+
+    def modify_acl_file_entry(self, file_name, entry, new_entry):
+        '''
+        Deascription:
+            Modify the acl_file acl list entry.
+        Args:
+            file_name: file name.
+            entry: acl entry to be modified.
+            new_entry: new acl entry.
+        Return:
+            none.
+        '''
+        acl_file = open(file_name, "r+")
+        new_permissions = ""
+        for line in acl_file.readlines():
+            line = line.split("\n")[0]
+            if line == entry:
+                line = new_entry
+                self.log.info("==>replaceing \n{}  with\n{}".
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments